transport/compress.go

```diff
@@ -0,0 +1,172 @@
+package transport
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"io"
+
+	"github.com/mmatczuk/scylla-go-driver/frame"
+
+	"github.com/klauspost/compress/s2"
+	"github.com/pierrec/lz4/v4"
+)
+
+var (
+	ErrComprCorrupt     = errors.New("compression: corrupt input")
+	ErrComprUnsupported = errors.New("compression: unsupported input")
+)
+
+type compr struct {
+	buf    []byte
+	tmp    []byte
+	header []byte
+
+	compressor func(int) (int, error)
+	c          lz4.Compressor
+}
+
+func newCompression(compress, snappy bool, bufSize int) *compr {
+	c := &compr{
+		buf: make([]byte, bufSize),
+		tmp: make([]byte, bufSize),
```

Comment:
This should probably be larger / smaller depending on compression type.
Before you calculated the size the temporary buffer.